pkg/sync/service.go

```
@@ -232,6 +237,66 @@
+				os.RemoveAll(tmpRawFile)
+			}
+		}()
+
+		stat, err := os.Stat(downloadFile)
+		if err != nil {
+			err = errors.Wrapf(err, "failed to stat the download file %v", downloadFile)
+			return
+		}
+
+		src, err := os.Open(downloadFile)
+		defer src.Close()
```

golangci-lint (1.62.2)

SA5001: should check error returned from os.Open() before deferring src.Close()

(staticcheck)